<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="local_planner_params_file"/>
    <arg name="global_planner_params_file"/>
    <arg name="nav_params_file"/>
    <arg name="map"/>
    <arg name="output" default="screen"/>
    <arg name="cmd_vel_topic" default="/cmd_vel" />
    <arg name="odom_topic" default="odom" />
    <arg name="model" default="turtlebot3_burger" />
    <param name="bool_goal_reached" value="true" />
    
    <node
        name="move_base"
        pkg="move_base"
        type="move_base"
        respawn="false"
        required="true"
        output="$(arg output)">
       
	<param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
	<rosparam file="$(find local_planning_performance_modelling)/config/$(arg model)/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find local_planning_performance_modelling)/config/$(arg model)/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find local_planning_performance_modelling)/config/$(arg model)/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find local_planning_performance_modelling)/config/$(arg model)/global_costmap_params.yaml" command="load" />
        
        <!-- planner params -->
    	<rosparam file="$(find local_planning_performance_modelling)/config/$(arg model)/base_local_planner_params.yaml" command="load" />
    	<rosparam file="$(find local_planning_performance_modelling)/config/$(arg model)/dwa_local_planner_params.yaml" command="load" />
        
        <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    	<remap from="odom" to="$(arg odom_topic)"/>
    	<param name="controller_frequency" value="20.0" /> <!-- Diese 3 einziger unterschied zu tb3-->
    	<param name="controller_patience" value="15.0" /> 
    	<param name="planner_frequency" value="1.0" /> <!-- default is 0.0 (plan only when needed)-->

    </node>

    <node
        name="map_server"
        pkg="map_server"
        type="map_server"
        args="$(arg map)"
        respawn="false"
        required="true"
        output="$(arg output)">
    </node> 


    <include file="$(find local_planning_performance_modelling)/launch/intermediate_planner.launch"/>
    
</launch>
